src/pages/MessageWall.jsx

```diff
@@ -112,6 +146,18 @@ function MessageWall(profile) {
       </div>
     );
   }
+  return (
+    <div>
+      {console.log(messages)}
+      <h3>Message Wall</h3>
+      {
+      messages.filter((message) => (message.pinned === true && message.serviceArea[0] === serviceArea.toLowerCase() && (message.target[0] === role.toLowerCase() || (message.target.length > 1 && message.target[1] === role.toLowerCase())))).map((message) => <Message key={message.id} id={message.id} title={message.title} body={message.body} updateMessages={updateMessages} />)
```

Comment:
I would probably change this to service area is in the service area array, instead of checking the first one. Similar with target.